FAQ.md

```diff
@@ -31,13 +42,34 @@ Supernode will maintain a bitmap which records the correspondence between peers
 
 **NOTE**: The scheduler will decide whether to download from the supernode or other peers. As for the detail of the scheduler, please refer to [scheduler algorithm](#what-is-the-peer-scheduling-algorithm-by-default)
 
+## What is the size of block(piece) when distribution
+
+Dragonfly tries to make block(piece) size dynamically to ensure efficiency.
+
+The size of pieces which is calculated as per the following strategy:
+
+- If file's total size is less than 200MB, then the piece size is `4MB` by default.
+- Otherwise, it equals to `min{ totalSize/100MB + 2 MB, 15MB }`.
+
+## What is the peer scheduling algorithm by default
+
+- Distribute the number of pieces evenly. Select the pieces with the smallest number in the entire P2P network so that the distribution of each piece in the P2P network is balanced to avoid "Nervous resources".
+
+- Nearest distance priority. For a peer, the piece closest to the piece currently being downloaded is preferentially selected, so that the peer can achieve the effect of sequential read and write approximately, which will improve the I/O efficiency of file.
+
+- Local blacklist and global blacklist. An example is easier to understand: When peer A fails to download from peer B, B will become the local blacklist of A, and then the download tasks of A will filter B out; When the number of failed downloads from B reaches a certain threshold, B will become the global blacklist, and all the download tasks will filter B out.
+
+- Self-isolation. PeerA will only download files from the supernode after it fails to download multiple times from other peers, and will also be added to the global blacklist. So the peer A will no longer interact with peers other than the supernode.
+
+- Peer load balancing. This mechanism will control the number of upload pieces and download pieces that each peer can provide simultaneously and the priority as the target peer.
+
 ## How do supernode and peers manage file cache which is ready for other peer's pulling
 
-Supernode will download files and cache them via CDN. For more information, please refer to [The sequence of supernode's CDN functionality](#what-is-the-sequence-of-supernode's-cdn-functionality).
+For supernode, it will download files and cache them via CDN. For more information, please refer to [The sequence of supernode's CDN functionality](#what-is-the-sequence-of-supernode's-cdn-functionality).
 
-After finishing distributing file from other peers, `dfget` should do two kinds of things:
+For one peer, after finishing distributing file from other peers, `dfget` will do these things:
 
-- first, construct all the pieces into unioned file(s);
+- first, construct all the pieces into a unioned file(s);
```

Comment:
an

Reply:
I think it  should be `a` rather than `an`.  @yeya24